### PR TITLE
Added handling of global phases

### PR DESCRIPTION
**Pull request by pehamTom**
This PR allows the ZXChecker to distinguish true equivalence and equivalence up to a global phase.

### CODE REVIEW

File: src/checker/zx/ZXChecker.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include "zx/FunctionalityConstruction.hpp"
 
 #include <chrono>
+#include <iostream>
```
Comment (burgholzer) [inline src/checker/zx/ZXChecker.cpp:16]: is this additional import necessary?

**Review by burgholzer — CHANGES_REQUESTED** [review]
Thanks for this PR!

It seems that some submodules got messed up during the process which is the main cause for the build failures and the messages reported by clang-tidy. This should probably fix itself once the QFR PR is merged and this PR points the QFR to the proper submodule commit.

The other failures are mostly related to a setuptools update. The corresponding fix is being tracked in #112.

Other than that, this looks good to me. Just a small request: could you maybe add an additional test for the ZXChecker, where circuits are only equivalent up to global phase? (could be something similar to the test added in the QFR repo).

File: src/checker/zx/ZXChecker.cpp (hunk 1 above)
Comment (pehamTom) [reply src/checker/zx/ZXChecker.cpp:16]: No

**Review by burgholzer — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**